.github/workflows/build.yml

```diff
@@ -114,3 +114,37 @@ jobs:
       - run: ./tool/shared/deploy.sh --robots ok default
         env:
           FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+
+  deploy_preview:
+    needs: test
+    runs-on: ubuntu-latest
+    if: '${{ github.event.pull_request.head.repo.full_name == github.repository }}'
+    steps:
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # submodules param for checkout action to get site-shared, 'true' might be enough
+          submodules: "recursive"
+      - run: mkdir -p $TMP
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+          ruby-version: 2.6.5
+      - uses: dart-lang/setup-dart@v1.2
+        with:
+          sdk: stable
+      - name: Install node dependencies
+        # TOOD: caching requires a lockfile for node deps, so does npm ci.
+        run: npm install
+      - name: Install dart dependencies
+        run: dart pub get
+      - run: ./tool/shared/write-ci-info.sh -v
+      - run: ./tool/build.sh
+      - uses: FirebaseExtended/action-hosting-deploy@0e16177d01e10c4560245404f959e75a9c633533
+        with:
+          repoToken: '${{ secrets.GITHUB_TOKEN }}'
+          firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_DART_DEV }}'
```

Comment:
This seems to require an exported service account key which will need an exemption from go/cute-sake.

Reply:
This was completed earlier